services/storage/src/simcore_service_storage/simcore_s3_dsm_utils.py

```diff
@@ -119,3 +119,20 @@ async def _get_fmd(
     directory_file_id_fmd = await _get_fmd(conn, directory_file_id)
 
     return directory_file_id if directory_file_id_fmd else None
+
+
+async def find_enclosing_file(
+    conn: SAConnection, user_id: UserID, file_id: SimcoreS3FileID
+) -> FileMetaDataAtDB | None:
+    kwnon_files = {
+        Path(known_file.file_id): known_file
+        for known_file in await db_file_meta_data.list_fmds(conn, user_id=user_id)
```

Comment:
here you are listing all the files of 'user_id'. that can be very heavy.
you have as argument the file_id, which at least contains the `project_id`, and possibly the `node_id` (if it is not a PublicAPI file), so you should be able to filter by them as well.

Reply:
Ok, I will have a look. I was not aware of these "conventions".